Fails to build Octave with Intel compiler due to C++ to Fortran calling convention issues

Spack fails to build Octave with Intel compiler due to C++ to Fortran calling convention issues.

### Expected Result

Builds Octave with Intel compiler.

### Actual Result

Fails to build Octave with Intel compiler.

### Steps to reproduce the issue

```console
$ spack spec octave+qt~readline@4.2.1 %intel@17.0.5 ^openblas threads=openmp
...
$ spack --restage install octave+qt~readline@4.2.1 %intel@17.0.5 ^openblas threads=openmp
...
==> Installing octave
==> Using cached archive: /lustre/home/rpm/spack/var/spack/cache/octave/octave-4.2.1.tar.gz
==> Already staged octave-4.2.1-o3lrrnoctonlzgx4wgbsyfqnqciabj3l in /lustre/home/rpm/spack/var/spack/stage/octave-4.2.1-o3lrrnoctonlzgx4wgbsyfqnqciabj3l
==> Already patched octave
==> Building octave [AutotoolsPackage]
==> Executing phase: 'autoreconf'
==> Executing phase: 'configure'
==> Error: ProcessError: Command exited with status 1:                                                            
    '/lustre/home/rpm/spack/var/spack/stage/octave-4.2.1-o3lrrnoctonlzgx4wgbsyfqnqciabj3l/octave-4.2.1/configure' '
--prefix=/lustre/spack/sandybridge/linux-centos7-x86_64/intel-17.0.5/octave-4.2.1-o3lrrnoctonlzgx4wgbsyfqnqciabj3l'
 '--with-blas=-L/lustre/spack/sandybridge/linux-centos7-x86_64/intel-17.0.5/openblas-0.2.20-247hkbjuggrspe2oyvq2smt
inigdpvml/lib -lopenblas' '--with-lapack=-L/lustre/spack/sandybridge/linux-centos7-x86_64/intel-17.0.5/openblas-0.2
.20-247hkbjuggrspe2oyvq2smtinigdpvml/lib -lopenblas' '--disable-readline' '--without-arpack' '--without-curl' '--wi
thout-fftw3' '--without-fftw3f' '--without-fltk' '--without-glpk' '--without-magick' '--without-hdf5' '--disable-ja
va' '--without-opengl' '--without-framework-opengl' '--without-qhull' '--without-qrupdate' '--without-z'

1 error found in build log:
     [ ... ]
     248   checking how to hardcode library paths into programs... immediate
     249   checking whether a usable Fortran compiler was found... yes
     250   checking how to get verbose linking output from /lustre/home/rpm/spack/lib/spack/env/intel/ifort... -v
     251   checking for Fortran 77 libraries of /lustre/home/rpm/spack/lib/spack/env/intel/ifort...  -L/lustre/spac
k/tools/linux-centos7-x86_64/gcc-4.8.5/intel-parallel-studio-cluster.2017.5-geurbg7kvebddctqiaqjqjoy2hzp45k5/compil
ers_and_libraries_2017.5.239/linux/compiler/lib/intel64_lin -L/usr/lib/gcc/x86_64-redhat-linux/4.8.5/ -L/usr/lib/gc
c/x86_64-redhat-linux/4.8.5/../../../../lib64 -L/usr/lib/gcc/x86_64-redhat-linux/4.8.5/../../../../lib64/ -L/lib/..
/lib64 -L/lib/../lib64/ -L/usr/lib/../lib64 -L/usr/lib/../lib64/ -L/usr/lib/gcc/x86_64-redhat-linux/4.8.5/../../../
 -L/lib64 -L/lib/ -L/usr/lib64 -L/usr/lib -limf -lm -lifport -lifcoremt -lsvml -lipgo -lirc -lpthread -lirc_s -ldl
     252   checking for dummy main to link with Fortran 77 libraries... none
     253   checking for Fortran 77 name-mangling scheme... lower case, underscore, no extra underscore
  >> 254   configure: error: to build Octave, the C++ to Fortran calling convention must be known.

See build log for details:
  /lustre/home/rpm/spack/var/spack/stage/octave-4.2.1-o3lrrnoctonlzgx4wgbsyfqnqciabj3l/octave-4.2.1/spack-build.out
```
[spack-spec.txt](https://github.com/spack/spack/files/1544863/spack-spec.txt)
[spack-build.env.txt](https://github.com/spack/spack/files/1544864/spack-build.env.txt)
[spack-build.out.txt](https://github.com/spack/spack/files/1544865/spack-build.out.txt)

### Information on your system

I am using CentOS 7.

--- a/var/spack/repos/builtin/packages/octave/package.py
+++ b/var/spack/repos/builtin/packages/octave/package.py
@@ -305,9 +305,8 @@ class Octave(AutotoolsPackage, GNUMirrorPackage):
             spec.satisfies('^intel-parallel-studio+mkl+ilp64')):
             config_args.append('F77_INTEGER_8_FLAG=-fdefault-integer-8')
 
-        # Use gfortran calling-convention %fj
-        if spec.satisfies('%fj'):
-            config_args.append('--enable-fortran-calling-convention=gfortran')
+        # Use gfortran calling-convention
+        config_args.append('--enable-fortran-calling-convention=gfortran')
 
         # Make sure we do not use qtchooser
         config_args.append('ac_cv_prog_ac_ct_QTCHOOSER=')

